pkg/store/connection/connection_recorder.go

```diff
@@ -131,6 +131,44 @@ func (c *Recorder) SaveNamespaceThreadID(threadID, namespace, connectionID strin
 	return c.transientStore.Put(getNamespaceKeyPrefix(prefix)(key), []byte(connectionID))
 }
 
+// RemoveConnection removes connection record from the store for given id
+func (c *Recorder) RemoveConnection(connectionID string) error {
+	record, err := c.GetConnectionRecord(connectionID)
+	if err != nil {
+		return err
+	}
+
+	if err = c.transientStore.Delete(getConnectionKeyPrefix()(connectionID)); err != nil {
+		return fmt.Errorf("unable to delete connection record from the transient store: connectionid=%s err=%w",
+			connectionID, err)
+	}
+
+	// remove connection records for different states from transient store
+	err = removeConnectionsForStates(c, connectionID)
+	if err != nil {
+		return err
+	}
+
+	err = c.store.Delete(getConnectionKeyPrefix()(connectionID))
+	if err != nil {
+		return fmt.Errorf("unable to delete connection record from the store: connectionid=%s err=%w", connectionID, err)
+	}
+
+	err = c.store.Delete(getDIDConnMapKeyPrefix()(record.MyDID, record.TheirDID))
+	if err != nil {
+		return fmt.Errorf("unable to delete did mapping connection record from the store: connectionid=%s err=%w",
+			connectionID, err)
+	}
+
+	// remove namespace, threadID and connection ID mapping from transient store
+	err = removeMappings(c, record)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
```

Comment:
can be simplified 

```
return removeMappings(c, record)
```

Reply:
Probably it is better to wrap this error if it will be already simplified error return inside _removeMappings_ function.

Reply:
I am ok with a `wrap` or `simplification`